vscode/microsoft-kiota/src/handlers/workspaceGenerationContextHandler.ts

```diff
@@ -0,0 +1,13 @@
+import { ClientOrPluginProperties } from "../kiotaInterop";
+
+interface WorkspaceGenerationContext {
+  clientOrPluginKey: string;
+  clientOrPluginObject: ClientOrPluginProperties;
+  generationType: string;
+}
+
+let workspaceGenerationContext: WorkspaceGenerationContext;
```

Comment:
I don't know how I feel about using a global variable here. Shouldn't this be passed to functions that need it as a context parameter?

i.e.

```typescript
function foo(generationContext: WorkspaceGenerationContext) {
    // .. do something with the context
}
```

I say this because I believe using a context object makes it easier to reason about dependencies for the function and allows for better testability. i.e. I can provide a dummy context for my tests.

Reply:
Are you suggesting that the context object be passed to the `execute` function?

Reply:
That's right.